src/masternodes/rpc_accounts.cpp

```diff
@@ -1203,6 +1203,72 @@ UniValue listaccounthistory(const JSONRPCRequest& request) {
     return slice;
 }
 
+UniValue getaccounthistory(const JSONRPCRequest& request) {
+    auto pwallet = GetWallet(request);
+
+    RPCHelpMan{"getaccounthistory",
+               "\nReturns information about account history.\n",
+               {
+                    {"owner", RPCArg::Type::STR, RPCArg::Optional::NO,
+                        "Single account ID (CScript or address)."},
+                    {"blockHeight", RPCArg::Type::NUM, RPCArg::Optional::NO,
+                        "Height to iterate from (downto genesis block)."},
+                    {"txn", RPCArg::Type::NUM, RPCArg::Optional::NO,
+                        "for order in block."},
+               },
+               RPCResult{
+                       "{}  An object with account history information\n"
+               },
+               RPCExamples{
+                       HelpExampleCli("getaccounthistory", "mxxA2sQMETJFbXcNbNbUzEsBCTn1JSHXST 103 2")
+                       + HelpExampleCli("getaccounthistory", "mxxA2sQMETJFbXcNbNbUzEsBCTn1JSHXST, 103, 2")
+               },
+    }.Check(request);
+
+    std::string ownerHexStr = request.params[0].getValStr();
+    CScript owner = DecodeScript(ownerHexStr);
+    uint32_t blockHeight = request.params[1].get_int();
+    uint32_t txn = request.params[2].get_int();
+
+    if (!paccountHistoryDB) {
+        throw JSONRPCError(RPC_INVALID_REQUEST, "-acindex is needed for account history");
+    }
+
+    pwallet->BlockUntilSyncedToCurrentChain();
+
+    LOCK(cs_main);
+    CCustomCSView view(*pcustomcsview);
+    CCoinsViewCache coins(&::ChainstateActive().CoinsTip());
+
+    UniValue result(UniValue::VOBJ);
+
+    auto shouldContinueToNextAccountHistory = [&](AccountHistoryKey const & key, CLazySerialize<AccountHistoryValue> valueLazy) -> bool {
+        if (owner == key.owner && blockHeight == key.blockHeight && txn == key.txn) {
+            const auto & value = valueLazy.get();
+            result = accounthistoryToJSON(key, value);
+        }
+        return false;
+    };
+
+    AccountHistoryKey startKey{owner, blockHeight, txn};
+
+    // revert previous tx to restore account balances to maxBlockHeight
+    paccountHistoryDB->ForEachAccountHistory([&](AccountHistoryKey const & key, AccountHistoryValue const & value) {
+        if (startKey.blockHeight > key.blockHeight) {
+            return false;
+        }
+        if (owner != key.owner) {
+            return false;
+        }
+        CScopeAccountReverter(view, key.owner, value.diff);
+        return true;
+    }, {owner, std::numeric_limits<uint32_t>::max(), std::numeric_limits<uint32_t>::max()});
```

Comment:
looking at the `listaccounthistory` implementation, this is only one piece of code for adding pool rewards to the output. by default `listaccounthistory` adds rewards to the output. do we need rewards history also via `getaccounthistory` @fuxingloh ? if yes, we might have to add other piece of the code also, and if not, we can remove this I think.

Reply:
We don't need pool reward, since you can't get pool reward by specifying the keys.